coroutines/devs/jess/milestones.md

```diff
@@ -0,0 +1,49 @@
+# Milestones
+
+This GSoC project has five stages. Below, each stage is listed along with its sub-goals and expected time to completion.
+
+## Warm-up Tasks
+#### Allotted Time: 3 Weeks
+- Setup Coroutines fork and solve the specialization puzzle in the intro doc (2 days)
+- Understand and document all test cases inside Coroutines repo (1 day)
+- Add missing test cases (1 day)
+- Investigate Async/Await tests mentioned in [this issue](https://github.com/storm-enroute/coroutines/issues/15), and port an Async/Await test into Coroutines (1 day)
+- Port 1/4 tests from Async/Await into Coroutines (1 day)
+- Port 1/4 tests from Async/Await into Coroutines (1 day)
+- Port 1/4 tests from Async/Await into Coroutines (1 day)
+- Port 1/4 tests from Async/Await into Coroutines (1 day)
+- Fix issues discovered while porting the tests over (1-5 days)
+
+## Async/Await
+#### Allotted Time: 2 Weeks
+- Write a design doc that contains a plan for the implementation of Async/Await. Make sure it pays special attention to failure handling (2 days)
+- Add and document motivating examples (1 day)
+- Implement Async/Await. (5 days) (**Note**: More details will be added here when they are planned out via the design doc)
+- Port remaining example and test code from Scala Async into Coroutines (2 days)
+
+## Enumerators
+#### Allotted Time: 3 Weeks
+- Write a design doc, including descriptions of any basic data structures to implement (2 days)
+- Create and document motivating examples for when enumerators are better than iterators (2 days)
+- Implement basic enumerators using coroutines (3 days) (**Note**: More details will be added here when they are planned out via the design doc)
+- Specialize enumerators (2 days)
+- Implement data structures described in the design doc (2 days)
+- Add boxing tests (2 days)
+- Add performance benchmarks (2 days)
+
+## Continuations
+#### Allotted Time: 3 Weeks
+- Write a deisgn doc (2 days)
+- Create and document motivating examples for continuations. Explain why, in some situations, they are better than coroutines (2 days)
```

Comment:
Include some time to read this: https://en.wikipedia.org/wiki/Delimited_continuation

And: http://users.eecs.northwestern.edu/~clk800/rand-test-study/_ifpdcbatsc/ifpdcbatsc-2009-10-8-12-02-00.pdf